### DISCUSSION TITLE
DataGrid -  how do I disable the data grid cell border when selected (visually allow row selection but not cell selection)?

### DISCUSSION THREAD

**Discussion by zaminda** [AvaloniaUI/Avalonia#8854] [Q&A] [answered]
How do I get rid of the border around individual cells in a DataGrid when they're selected? I would like to have row selection, not individual cell selection (appearance). Thank you :)

**Comment by timunie** [comment, marked as answer]
Try to override this Selector: https://github.com/AvaloniaUI/Avalonia/blob/62a5ee14fb12b26672ea25a06b2154b19fd83a01/src/Avalonia.Controls.DataGrid/Themes/Fluent.xaml#L162-L164

Happy coding
Tim

**Reply by zaminda** [reply]
Brilliant! Thank you very much

**Reply by zaminda** [reply]
@timunie  BTW IsVisible has to be False isn't it?

**Reply by timunie** [reply]
> @timunie  BTW IsVisible has to be False isn't it? 

I think so. Haven't tried it on my own.

**Reply by aloneguid** [reply]
Didn't make a difference for me :(

**Reply by KNPDesign** [reply]
@aloneguid 

Tested it, 	
<Style Selector="DataGrid:focus DataGridCell:current /template/ Grid#FocusVisual">
	<Setter Property="IsVisible" Value="False" />
</Style>
	
	Is definitely the correct solution

**Reply by AydinBiber-Dekker** [reply]
If you're like me and haven't overridden a Style before, here is how you do it: https://docs.avaloniaui.net/docs/basics/user-interface/styling/styles

**Reply by MeltyPlayer** [reply]
Just following up on this, I had more luck with:

```
<DataGrid>
  <DataGrid.Styles>
    <Style Selector="DataGridCell:current /template/ Grid#FocusVisual"> 
       <Setter Property="IsVisible" Value="false" /> 
     </Style> 
  </DataGrid.Styles>

  ...
</DataGrid>
```

**Reply by DmitriiShamrikov** [reply]
How do I do this if I use Semi Avalonia theme?

**Reply by rabbitism** [reply]
> How do I do this if I use Semi Avalonia theme?

should be quite similar

**Reply by DmitriiShamrikov** [reply]
It doesn't work and I can't figure out from the theme source what needs to be different. Anyone had any success?

**Reply by rabbitism** [reply]
> It doesn't work and I can't figure out from the theme source what needs to be different. Anyone had any success?
@DmitriiShamrikov 
```xaml
        <Style Selector="DataGridCell:focus /template/ Panel#FocusVisual">
            <Setter Property="IsVisible" Value="False"/>
        </Style>
        <Style Selector="DataGridCell:current /template/ Rectangle#CurrencyVisual">
            <Setter Property="IsVisible" Value="False" />
        </Style>
```

**Reply by DmitriiShamrikov** [reply]
Great, thanks, @rabbitism !

**Reply by Br1Z** [reply]
It worked for me.
```xaml
<Style Selector="DataGridCell:focus /template/ Grid#FocusVisual">
    <Setter Property="IsVisible" Value="False"/>
</Style>
<Style Selector="DataGridCell:current /template/ Rectangle#CurrencyVisual">
    <Setter Property="IsVisible" Value="False"/>
</Style>
```
